### PR TITLE
Implemented Geo Point

### PR DESCRIPTION
**Pull request by matriv**

### CODE REVIEW

**Comment by matriv** [comment]
@aelesbao Plz review

File: sql/src/main/java/io/crate/protocols/postgres/types/PGGeoPointType.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    public Object readBinaryValue(ChannelBuffer buffer, int valueLength) {
+        assert valueLength == TYPE_LEN : "length should be " + TYPE_LEN + " because point is 2*64 bit double. " +
+                                         "Actual length: " + valueLength;
+        return new Double[] {buffer.readDouble(), buffer.readDouble()};
+    }
+
+    @Override
+    byte[] encodeAsUTF8Text(@Nonnull Object array) {
+        Double[] values = (Double[]) array;
+        List<Byte> encodesValues = new ArrayList<>();
```
Comment (aelesbao) [inline sql/src/main/java/io/crate/protocols/postgres/types/PGGeoPointType.java]: `encodedValues`

File: sql/src/main/java/io/crate/protocols/postgres/types/PGGeoPointType.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        encodesValues.add((byte)'(');
+
+        for (int i = 0; i < values.length; i++) {
+            double d = values[i];
+            if (i > 0) {
+                encodesValues.add((byte) ',');
+            }
+            byte[] bytes = DoubleType.INSTANCE.encodeAsUTF8Text(d);
+            for (byte aByte : bytes) {
+                encodesValues.add(aByte);
```
Comment (aelesbao) [inline sql/src/main/java/io/crate/protocols/postgres/types/PGGeoPointType.java]: can you use `addAll` here?

**Comment by aelesbao** [comment]
lgtm otherwise

**Comment by matriv** [comment]
@mfussenegger Plz review

File: sql/src/main/java/io/crate/protocols/postgres/types/PGGeoPointType.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+    @Override
+    public Object readBinaryValue(ChannelBuffer buffer, int valueLength) {
+        assert valueLength == TYPE_LEN : "length should be " + TYPE_LEN + " because point is 2*64 bit double. " +
+                                         "Actual length: " + valueLength;
+        return new Double[] {buffer.readDouble(), buffer.readDouble()};
+    }
+
+    @Override
+    byte[] encodeAsUTF8Text(@Nonnull Object array) {
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/protocols/postgres/types/PGGeoPointType.java]: would it make sense to create static variants of the Functions in PGArray which take the innerType as additional argument? This is basically the same.

File: sql/src/test/java/io/crate/integrationtests/PostgresITest.java [hunk 4]
```diff
@@ -131,6 +132,28 @@ public void testArrayTypeSupport() throws Exception {
     }
 
     @Test
+    public void testGeoPointSupport() throws Exception {
```
Comment (mfussenegger) [inline sql/src/test/java/io/crate/integrationtests/PostgresITest.java]: please remove the test, enabling `UseJdbc` on the existing geo_point specific integration tests should be enough

File: sql/src/test/java/io/crate/integrationtests/TransportSQLActionTest.java [hunk 5]
```diff
@@ -1440,12 +1441,11 @@ public void testGroupByOnIpType() throws Exception {
     }
 
     @Test
-    @UseJdbc(false) // geo_point support is missing
     public void testInsertAndSelectGeoType() throws Exception {
         execute("create table geo_point_table (id int primary key, p geo_point) with (number_of_replicas=0)");
         ensureYellow();
-        execute("insert into geo_point_table (id, p) values (?, ?)", new Object[]{1, new Double[]{47.22, 12.09}});
-        execute("insert into geo_point_table (id, p) values (?, ?)", new Object[]{2, new Double[]{57.22, 7.12}});
+        execute("insert into geo_point_table (id, p) values (?, ?)", new Object[]{1, new PGpoint(47.22, 12.09)});
```
Comment (mfussenegger) [inline sql/src/test/java/io/crate/integrationtests/TransportSQLActionTest.java]: why is this change necessary and how does this work if it isn't executed with jdbc but with the transport client?

File: sql/src/main/java/io/crate/protocols/postgres/types/PGGeoPointType.java (hunk 3 above)
Comment (matriv) [inline sql/src/main/java/io/crate/protocols/postgres/types/PGGeoPointType.java]: I have to use () instead of {}

**Comment by mfussenegger** [comment]
lgtm
